packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -2885,6 +2899,14 @@ const getRequestHelperFunctions = (
 
 				const tempRequestOptions = applyPaginationRequestData(requestOptions, paginateRequestData);
 
+				if (!validateUrl(tempRequestOptions.uri as string)) {
+					throw new NodeOperationError(node, `'${paginateRequestData.url}' is not a valid URL.`, {
+						itemIndex,
+						runIndex,
+						type: 'invalid_url',
```

Comment:
I would avoid extending NodeApiError interface for this special case, as this error would be replaced in Http node anyway we could send some technical message for identification

Reply:
I prefer using a `type` to check on instead of using the message. We also use this pattern for other errors (like ExpressionError)